### PR TITLE
(GH-674) Create tool manifest for restoring Cake.Tool

### LINKED ISSUE #674: Switch to using Cake.Tool global .NET Core tool
Since we already are changing all of the utilities that are being used to be .NET Core compatible, then maybe we should also update Cake.Recipe to build with .NET Core itself.

### PR DESCRIPTION
**Pull request by AdmiringWorm**
This pull request adds a minimalistic .NET Core tool manifest to allow the ability to use Cake.Tool using dotnet cake.
It also changes the normal build process to use cake global tool both locally, on appveyor and uses cake-action on github actions.

resolves #674

### CODE REVIEW

**Comment by gep13** [comment]
@AdmiringWorm I just went to run this on my Mac, and I get the following when trying to run it locally:

image

Any ideas on what is doing on here?

**Comment by AdmiringWorm** [comment]
@gep13 most likely, you are being hit by issue #626 I think.
At least it looks that way.

A bit odd that you got that error though, as it should be running the normal build if you don't use any arguments on the bootstrapper (assuming you are running the bootstrapper through git hooks)

File: build.sh [hunk 1]
```diff
@@ -32,92 +34,113 @@ CAKE_ARGUMENTS=()
...
+        fi
+    fi
+
+    # Restore tools from NuGet.
+    pushd "$TOOLS_DIR" >/dev/null
+    if [ ! -f "$PACKAGES_CONFIG_MD5" ] || [ "$(cat "$PACKAGES_CONFIG_MD5" | sed 's/\r$//')" != "$($MD5_EXE "$PACKAGES_CONFIG" | awk '{ print $1 }')" ]; then
+        find . -type d ! -name . ! -name 'Cake.Bakery' | xargs rm -rf
+    fi
+
+    . $MONO_EXE "$NUGET_EXE" install -ExcludeVersion
```
Comment (AdmiringWorm) [inline build.sh:96]: Forget my previous comment, it seems to be that something is wrong with this call on OSX.
I tested it on Linux (arch linux), but there probably are some differences that I am not aware of between OSX and Linux in the way binaries can be call from a shell script

Comment (gep13) [reply build.sh:96]: Happy to try something out if you have some ideas.

Comment (AdmiringWorm) [reply build.sh:96]: Unfortunately, right now I got no clue at all what to do about it. 😢

Comment (gep13) [reply build.sh:96]: Ha ha :joy: Not a problem.  I have even less of a clue, since I rely heavily on @devlead when it comes to .sh scripts 😄

Comment (AdmiringWorm) [reply build.sh]: @gep13 can you give my latest changes a try.
I made some minor changes to the shell script that I believe should change the issue.

**Comment by AdmiringWorm** [comment]
Need to look into why the appveyor build fails, no idea why right now.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@AdmiringWorm thanks for making these changes!

**Pull request merged**